spec.html

```diff
@@ -14620,6 +14620,23 @@ <h1>TypedArray Exotic Objects</h1>
       <p>TypedArrays have the same internal slots as ordinary objects and additionally [[ViewedArrayBuffer]], [[ArrayLength]], [[ByteOffset]], [[ContentType]], and [[TypedArrayName]] internal slots.</p>
       <p>An object is a <dfn id="typedarray" oldids="integer-indexed-exotic-object" variants="TypedArrays">TypedArray</dfn> if its [[GetOwnProperty]], [[HasProperty]], [[DefineOwnProperty]], [[Get]], [[Set]], [[Delete]], and [[OwnPropertyKeys]] internal methods use the definitions in this section, and its other essential internal methods use the definitions found in <emu-xref href="#sec-ordinary-object-internal-methods-and-internal-slots"></emu-xref>. These methods are installed by TypedArrayCreate.</p>
 
+      <emu-clause id="sec-typedarray-preventextensions" type="internal method">
+        <h1>[[PreventExtensions]] ( ): a normal completion containing a Boolean</h1>
+        <dl class="header">
+          <dt>for</dt>
+          <dd>a TypedArray _O_</dd>
+        </dl>
+        <emu-alg>
+          1. Let _buffer_ be _O_.[[ViewedArrayBuffer]].
+          1. Let _taIsFixedLength_ be *true*.
+          1. If _O_.[[ArrayLength]] is ~auto~, set _taIsFixedLength_ to *false*.
```

Comment:
```suggestion
          1. If _O_.[[ArrayLength]] is ~auto~, return *false*.
```
etc, yeah? I don't see any reason for the intermediate variable. And then perhaps move the note above this step. Otherwise LGTM.

Reply:
I like the current version because it's in 2 (macro) steps: check if something is a fixed-length TA, then return based on that. Returning early with the alias still named _taIsFixedLength_ might raise questions from the reader like "wait, aren't AUTO TAs also not fixed length"?

WDYT about moving IsTypedArrayFixedLength into a predicate AO?